src/component/RegistrationForm/RegistrationForm.tsx

```diff
@@ -0,0 +1,125 @@
+    onErrorSet: (error: string) => void;
+    history: any;
+    errorType: string;
+}
+
+interface objForm {
+    name?: string;
+    email: string;
+    password: string;
+    file?: string;
+}
+
+const RegistrationForm: React.FC<RegistrationPanelProps> = ({ history, language, isSignIn, errorType, onLoginGet, onRegistrationSend, onErrorSet }) => {
+    const [fileInformation, setFileInformation] = useState('');
+    const [errorDescription, setErrorDescription] = useState('');
+
+    const getFile = useCallback((event: React.MouseEvent<HTMLLabelElement>): void => {
+        const target = event.target as HTMLInputElement;
+        console.log(target.tagName);
+        if (target.tagName === 'INPUT') {
+            const file = target.files![0].name;
+            setFileInformation(file);
+        }
+    }, []);
+
+    const onSubmitClick = useCallback((event: React.MouseEvent<HTMLFormElement>): void => {
+        event.preventDefault();
+        const target = event.target as HTMLFormElement;
+        const formData = new FormData(target);
+        let flagError = false;
+        const objFormData = {} as objForm;
+        for (let pair of formData.entries()) {
+            if (pair[1] === '' && pair[0] !== 'file') {
+                flagError = true;
+                break;
+            } else if (pair[0] !== 'file') {
+                if (pair[0] === 'email') {
+                    objFormData.email = pair[1] as string;
+                } else if (pair[0] === 'password') {
+                    objFormData.password = pair[1] as string;
+                } else if (pair[0] === 'name') {
+                    objFormData.name = pair[1] as string;
+                }
+            }
+        }
+        if (flagError) {
+            onErrorSet('Empty');
+        } else {
+            if (isSignIn) {
+                onRegistrationSend(objFormData, true);
+            } else {
+                onLoginGet(objFormData, true);
+            }
+        }
+    }, [onLoginGet, onRegistrationSend, onErrorSet, isSignIn]);
+
+    useEffect(() => {
+        if (errorType === 'Not found') {
+            setErrorDescription(languages.errorSpanFound[language]);
+        } else if (errorType === 'Empty') {
+            setErrorDescription(languages.errorSpanEmpty[language]);
+        } else if (errorType === 'Exist') {
+            setErrorDescription(languages.errorSpanExist[language]);
+        } else if (errorType === 'Success') {
+            onErrorSet('');
+            history.push('/');
+        }
+    }, [language, onErrorSet, history, errorType]);
+
+    return (
+        <div className={styles.formContainer}>
+            <form className={styles.form} onSubmit={ onSubmitClick } method="POST">
+                <input className={styles.formElement} type="email" name="email" placeholder={ language === 'ru' ? 'Введите вашу почту' : language === 'en' ? 'Enter your email' : 'Entrer votre email' } />
```

Comment:
you could try react intl packages for that:

https://www.npmjs.com/package/react-intl
https://github.com/i18next/react-i18next